elasticdl/python/tests/pserver_servicer_test.py

```diff
@@ -240,108 +240,167 @@ def test_pull_embedding_vector(self):
+                    expected_value,
+                    self._parameters.non_embedding_params[name].numpy(),
+                )
             )
 
+        actual_embed_table = self._parameters.get_embedding_param(
+            self._embedding_info.name, range(len(self.expected_embed_table))
+        )
+        self.assertTrue(
+            np.allclose(self.expected_embed_table, actual_embed_table)
+        )
+
         # Test applying gradients with same name
+        for name, var in zip(self.var_names, self.var_values):
+            self._parameters.non_embedding_params[name] = tf.Variable(var)
         req = elasticdl_pb2.PushGradientRequest()
-        for g in grad_value:
-            emplace_tensor_pb_from_ndarray(req.gradients, g, name=var_name[0])
+        for g in self.grad_values1:
+            emplace_tensor_pb_from_ndarray(
+                req.gradients, g, name=self.var_names[0]
+            )
         res = self._stub.push_gradient(req)
         self.assertEqual(res.accepted, True)
         self.assertEqual(res.model_version, 2)
         expected_values = [
-            var_value[0]
-            - self._lr * grad_value[0] * 2
-            - self._lr * grad_value[1],
-            var_value[1] - self._lr * grad_value[1],
+            self.var_values[0]
+            - self._lr * self.grad_values1[0]
+            - self._lr * self.grad_values1[1],
+            self.var_values[1],
         ]
-        for expected_value, name in zip(expected_values, var_name):
+        for expected_value, name in zip(expected_values, self.var_names):
             self.assertTrue(
-                np.allclose(expected_value, non_embedding_params[name].numpy())
+                np.allclose(
+                    expected_value,
+                    self._parameters.non_embedding_params[name].numpy(),
+                )
             )
 
-        # TODO: test applying gradients with embedding params
-
     def test_push_gradient_sync_update(self):
         self.create_server_and_stub(
             grads_to_wait=2, lr_staleness_modulation=False, use_async=False
         )
-        var_name = ["test_1", "test_2"]
-        var_value = [
-            np.array([10.0, 20.0, 30.0], np.float32),
-            np.array([20.0, 40.0, 60.0], np.float32),
-        ]
-
-        non_embedding_params = self._parameters.non_embedding_params
-        for i in range(2):
-            non_embedding_params[var_name[i]] = tf.Variable(var_value[i])
-
-        grad_value0 = [
-            np.array([1.0, 2.0, 3.0], np.float32),
-            np.array([2.0, 4.0, 6.0], np.float32),
-        ]
-        grad_value1 = [
-            np.array([0.0, 0.0, 7.0], np.float32),
-            np.array([9.0, 9.0, 6.0], np.float32),
-        ]
+        self.push_gradient_test_setup()
 
         req = elasticdl_pb2.PushGradientRequest()
         req.model_version = 0
-        for g, name in zip(grad_value0, var_name):
+        for g, name in zip(self.grad_values0, self.var_names):
```

Comment:
To cover more code and make the test more robust, it's better to add an embedding gradient here.

```python
self.embedding_grads_1 = tf.IndexedSlices(
            values=np.random.rand(3 * dim)
            .reshape((3, dim))
            .astype(np.float32),
            indices=(2, 2, 3),
        )
```